non_semantic_speech_benchmark/distillation/models.py

```diff
@@ -28,64 +30,90 @@ def _sample_to_features(x):
-                    mobilenet_size='small',
-                    frontend=True,
-                    avg_pool=False):
-  """Make a keras student model."""
-
-  def _map_fn_lambda(x):
-    return tf.map_fn(_sample_to_features, x, dtype=tf.float64)
-
-  def _map_mobilenet_func(mnet_size):
-    mnet_size_map = {
-        'tiny': mobilenetv3_tiny,
-        'small': tf.keras.applications.MobileNetV3Small,
-        'large': tf.keras.applications.MobileNetV3Large,
-    }
-    assert mnet_size in mnet_size_map
-    return mnet_size_map[mnet_size.lower()]
-
-  if frontend:
-    model_in = tf.keras.Input((None,), name='audio_samples')
-    feats = tf.keras.layers.Lambda(_map_fn_lambda)(model_in)
-    feats.shape.assert_is_compatible_with([None, None, 96, 64])
-    feats = tf.transpose(feats, [0, 2, 1, 3])
-    feats = tf.reshape(feats, [-1, 96, 64, 1])
-  else:
-    model_in = tf.keras.Input((96, 64, 1), name='log_mel_spectrogram')
-    feats = model_in
-  model = _map_mobilenet_func(mobilenet_size)(
-      input_shape=[96, 64, 1],
-      alpha=alpha,
-      minimalistic=False,
-      include_top=False,
-      weights=None,
-      pooling='avg' if avg_pool else None,
-      dropout_rate=0.0)
-  model_out = model(feats)
-  if avg_pool:
-    model_out.shape.assert_is_compatible_with([None, None])
-  else:
-    model_out.shape.assert_is_compatible_with([None, 3, 2, None])
-  if bottleneck_dimension:
-    embeddings = tf.keras.layers.Flatten()(model_out)
-    embeddings = tf.keras.layers.Dense(
-        bottleneck_dimension, name='distilled_output')(
-            embeddings)
-  else:
-    embeddings = tf.keras.layers.Flatten(name='distilled_output')(model_out)
-  # TODO(joelshor): These final layers can be large. Investigate the compression
-  # techniques described in
-  # https://blog.tensorflow.org/2020/02/matrix-compression-operator-tensorflow.html?m=1
-  output = tf.keras.layers.Dense(
-      output_dimension, name='embedding_to_target')(
-          embeddings)
-  output_model = tf.keras.Model(inputs=model_in, outputs=output)
-
-  return output_model
+def _map_fn_lambda(x):
+  return tf.map_fn(_sample_to_features, x, dtype=tf.float64)
+
+
+def _make_bottleneck_block(bottleneck_dim, qat=False, compressor=None):
+  if bottleneck_dim:
+    if compressor is not None:
+      bottleneck = CompressedDense(
+        bottleneck_dim, compression_obj=compressor, name='distilled_output')
+    else:
+      bottleneck = tf.keras.layers.Dense(
+        bottleneck_dim, name='distilled_output')
+      if qat:
+        bottleneck = tfmot.quantization.keras.\
+          quantize_annotate_layer(bottleneck)
+    return bottleneck
+  # pass through
+  return tf.keras.layers.Activation(None, name='distilled_output')
+
+
+def _make_mobilenet_block(mnet_size, alpha, avg_pool):
+  mnet_size_map = {
+      'tiny': mobilenetv3_tiny,
+      'small': v3_util.MobileNetV3Small,
+      'large': v3_util.MobileNetV3Large,
+  }
+  assert mnet_size in mnet_size_map
```

Comment:
nit: Can you make this a ValueError exception instead of an assert?

Reply:
Done.